### PR TITLE
Add Ruby Tests

### PR DESCRIPTION
**Pull request by bheiskell**
I spent some time today figuring out how to wire up testing:

- Added rspec for the testing framework
- Set the default rake task to `test:test` which will initialize a test database
- Wired Grunt to execute shotgun in the background (inconjunction with watch)
- Wired Grunt to execute Rake to perform the ruby build
- Added a base status check controller test
- Added a user creation test (which utilizes the initiazed test database)
- Moved the db:seed task into the actual migrations (this removes the need for a separate rake call)

### CODE REVIEW

**Comment by bheiskell** [comment]
@blaskovicz 
I remembered you mentioned that it would be beneficial to get testing going. This should cover the foundations for defining tests against the sinatra end-points.

I'm not certain how you want to handle angular testing. If you plan to mock you data services, then you can run them whenever. However if you want end to end tests, that will be more complicated to get running.

**Comment by blaskovicz** [comment]
I can try to get the frontend testing mocked out.

I will try to take a look at this tomorrow.

I want to try to rebase my matches branch since some of the commits are random "checkpoint" commits.

File: Gruntfile.coffee [hunk 1]
```diff
@@ -57,12 +57,30 @@ module.exports = (grunt) ->
...
+          'spec/**/*.rb'
+          'Rakefile'
+          'config.ru'
+        ]
+        tasks: [
+          'bgShell:rake'
+        ]
+    bgShell:
+      rake:
+        cmd: 'bundler exec rake'
```
Comment (blaskovicz) [inline Gruntfile.coffee]: should this be the "bundle" command?

File: Gruntfile.coffee [hunk 2]
```diff
@@ -57,12 +57,30 @@ module.exports = (grunt) ->
...
+        fail: true
+      shotgun:
+        cmd: 'bundler exec shotgun config.ru'
```
Comment (blaskovicz) [inline Gruntfile.coffee]: should this be the "bundle" command?

File: db/init.rb [hunk 3]
```diff
@@ -1,10 +1,16 @@
...
-$db_url = ENV["DATABASE_URL"] || (
-  ENV["RACK_ENV"] == "production" ?
-  "postgres://localhost/magicstick" :
-  "sqlite://magicstick.db"
-)
-$DB = Sequel.connect($db_url)
 
+$db_url = ENV["DATABASE_URL"]
+
+case ENV["RACK_ENV"]
```
Comment (blaskovicz) [inline db/init.rb]: this can be cleaned up a bit to assign directly from the case statement.

**Comment by bheiskell** [comment]
@blaskovicz 

I've applied the suggested fixes and brought in the changes from merge request #6 into this one. Inlining 6's description:

This adds support for testing with Travis and pushing to Heroku. However, Travis is not configured to deploy to Heroku yet. I'm having trouble with the CLI (won't encrypt for me). It wouldn't matter anyways, because Travis should be configured with blaskovicz's account, because that's the authoritative repository. Here are the instructions:

http://docs.travis-ci.com/user/deployment/heroku/

I'm not certain how github will handle this merge request given that it has overlap with #5.

Also, if I guessed correctly that your Heroku account is your gmail address, then I can transfer ownership of that application to you.

**Comment by bheiskell** [comment]
@blaskovicz If you're going to be out of reach this weekend, please be sure to merge this request so I can work off the official branch :-)

**Pull request merged**